src/site/topic2-cpp.rst

```diff
@@ -0,0 +1,646 @@
+    * ``int`` is the function return type --- more on this later
+    * Although not necessary, we see ``return 0`` at the end
+        * This is the value that the ``main`` function is returning
+        * Typically ``0`` is an *exit code* that means things went well
+
+* The ``"Hello, World!"`` is simply the string being output
+
+* The ``std::cout`` is how we will print things out
+    * The ``std`` part is the namespace where ``cout`` is
+        * Like the ``math`` in ``math.sin`` in Python
+    * ``std`` is short for *standard*
+
+* ``std::endl`` is how we end the line we're printing
+    * We could simply use a new line character, but the benefit of using ``endl`` is that we flush the buffer
+
+* Like Java:
+    * You will also see the use of semicolons (``;``) at the end of the lines of code (we need these)
+    * We use squiggly brackets for starting end ending the function ``{ ... }``
+
+
+Variables and Types
+===================
+
+
+**Python**
+
+.. code-block:: python
+    :linenos:
+    :emphasize-lines: 2,3,4
+
+    # Python --- Declaring and assigning variables
+    anInt = 5
+    aFloat = 5.5
+    aString = "5"       # Could have used '5'
+
+
+**Java**
+
+.. code-block:: java
+    :linenos:
+    :emphasize-lines: 4,5,6
+
+    // Java --- Declaring and assigning variables
+    public class SomeClass {
+        public static void main(String[] args){
+            int anInt = 5;
+            double aFloat = 5.5;    // doubles now for float
+            String aString = "5";   // Mind the double quotes
+
+        }
+    }
+
+
+**C++**
+
+.. code-block:: cpp
+    :linenos:
+    :emphasize-lines: 1,7
+
+    #include <string>
+
+    // Cpp --- Declaring and assigning variables
+    int main() {
+        int anInt = 5;
+        double aFloat = 5.5;         // doubles now for float
+        std::string aString = "5";   // Mind the double quotes
+        return 0;
+    }
+
+* You will see that C++ and Java are very similar here
+* The only catch is that, unlike Java and Python, we don't get strings for free out of the box
+* We can use strings by including ``<string>``
+    * Or ``<iostream>`` since we get strings that that too
+* Also note the use of ``std`` before the string type
+
+* Just like Java, with C++ you cannot swap the types we store in our variables like we could in Python
+    * Although, there kinda' is a way to do this in C++, but we will ignore that
+
+* Again, like Java, with C++ we can declare a variable on a single like and then assign it a value later
+
+
+Constants
+---------
+
+**Python**
+
+.. code-block:: python
+    :linenos:
+
+    # Python --- Constants
+    SALES_TAX = 1.15    # This is a constant. Leave this alone!
+
+
+**Java**
+
+.. code-block:: java
+    :linenos:
+
+    // Java --- Constants
+    final double SALES_TAX = 1.15;      // Java will get mad if we mess with this
+
+
+**C++**
+
+.. code-block:: cpp
+    :linenos:
+
+    // Cpp --- Constants
+    const double SALES_TAX = 1.15;
```

Comment:
Don't forget [`constexpr`](https://en.cppreference.com/w/cpp/language/constexpr).

Reply:
I prefer to forget it, here at least.